modules/hostap/src/supp_events.c

```diff
@@ -21,10 +145,22 @@ int send_wifi_mgmt_event(const char *ifname, enum net_event_wifi_cmd event, int
 
 	switch (event) {
 	case NET_EVENT_WIFI_CMD_CONNECT_RESULT:
-		wifi_mgmt_raise_connect_result_event(iface, status);
+		wifi_mgmt_raise_connect_result_event(iface, *(int *)status);
 		break;
 	case NET_EVENT_WIFI_CMD_DISCONNECT_RESULT:
-		wifi_mgmt_raise_disconnect_result_event(iface, status);
+		wifi_mgmt_raise_disconnect_result_event(iface, *(int *)status);
+		break;
+	case NET_EVENT_WPA_SUPP_CMD_INT_EVENT:
+		event_data.data = &data;
+		ret = process_status(&event_data, (char *)status);
+		if (ret >= 0) {
+			net_mgmt_event_notify_with_info(NET_EVENT_WPA_SUPP_INT_EVENT,
+						iface, &event_data, sizeof(event_data));
+		} else {
+			if (ret != -ENOTSUP) {
+				wpa_printf(MSG_ERROR, "NM event notification failed\n");
```

Comment:
if you can move this log `process_status` and modify to say `WPA supplicant event is not supported`, then we can remove this branch. Please use WPA_S/WPA_SUPP instead of NM, NM was meant to be used in upstream zephyr to cover different types of NMs.

Reply:
This is to keep the difference between events not supported and the events which are supported but their parsing failed. Will replace NM with WPA_SUPP.

Reply:
Sure, but my suggestion of moving this to `process_status` still applies.

Reply:
Right. Will remove this log. Eror case in `process_status` is already being logged.

Reply:
not for `-ENOTSUP`

Reply:
Logged the -ENOTSUP as well case in `process_status` .